src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageRestoreBar.xaml.cs

```diff
@@ -81,6 +82,7 @@ public PackageRestoreBar(INuGetSolutionManagerService solutionManager, IPackageR
                 if (_projectContextInfo?.ProjectStyle == ProjectModel.ProjectStyle.PackageReference)
                 {
                     _packageRestoreManager.AssetsFileMissingStatusChanged += OnAssetsFileMissingStatusChanged;
+                    _packageRestoreManager.SetCurrentBannerProjectIdInPMUI(_projectContextInfo.ProjectId);
```

Comment:
I'm not sure this is good API design. 
A PackageRestoreManager knowing about the banner project id seems weird.

Reply:
Yeah, I felt that too, I was looking for a way to be able to call `RaiseAssetsFileMissingEventForProjectAsync` from any restore and only check the project that has the missing assets, I will investigate more about events and try to make this better later.